File: Python3/1001 - 1200/1009 Complement of Base 10 Integer.py
# 32ms, 88.75%; 13.7MB, 6.67%
from math import log2

class Solution:
    def bitwiseComplement(self, N: int) -> int:
        if N == 0: return 1
        return 2 ** (int(log2(N)) + 1) - 1 - N
    

File: 1001 - 1200/1009 Complement of Base 10 Integer.cc
// 4ms, 51.01%; 8.4MB, 33.33%
class Solution
{
public:
    int bitwiseComplement(int N)
    {
        if (N == 0) return 1;
        return pow(2, 1 + int(log2(N))) - 1 - N;
    }
};
